bench/workgen/workgen.cpp

```diff
@@ -793,24 +875,51 @@ ContextInternal::create_all(WT_CONNECTION *conn)
 
     /* Walk the entries in the metadata and extract the dynamic set. */
     while ((ret = cursor->next(cursor)) == 0) {
-        const char *key, *value;
+        const char *key, *v;
         if ((ret = cursor->get_key(cursor, &key)) != 0) {
             THROW(
               "Error getting the key for a metadata entry while extracting dynamic set of tables.");
         }
-        if ((ret = cursor->get_value(cursor, &value)) != 0) {
+        if ((ret = cursor->get_value(cursor, &v)) != 0) {
             THROW(
               "Error getting the value for a metadata entry while extracting dynamic set of "
               "tables.");
         }
 
-        if (std::string(value).find(DYN_TABLE_APP_METADATA) != std::string::npos &&
-          WT_PREFIX_MATCH(key, "table:")) {
+        std::string value = std::string(v);
+        size_t pos = value.find(DYN_TABLE_APP_METADATA);
+        if (pos != std::string::npos && WT_PREFIX_MATCH(key, "table:")) {
             // Add the table into the list of dynamic set. We are single threaded here and hence
             // do not yet need to protect the dynamic table structures with a lock.
             _dyn_tint[key] = _dyn_tint_last;
             _dyn_table_names[_dyn_tint_last] = key;
             _dyn_table_runtime[_dyn_tint_last] = TableRuntime();
+
+            size_t start, end;
+            pos = value.find(MIRROR_TABLE_APP_METADATA);
+            // If the table has a mirror, add the mirror name to the runtime data.
+            if (pos != std::string::npos) {
+                start = value.find_first_of('=', pos);
+                end = value.find_first_of(",\"", start);
+                if (start == std::string::npos || end == std::string::npos) {
+                    THROW("Unable to retrieve mirror name from metadata.");
+                }
+                _dyn_table_runtime[_dyn_tint_last]._mirror =
```

Comment:
Does a mirror table also get a mirror table name (with _is_base as false ?

Reply:
Yes, the base table gets the name of the mirror and the mirror table gets the name of the base.